lib/sablon/image.rb

```diff
@@ -0,0 +1,17 @@
+module Sablon
+  class Image
+    include Singleton
```

Comment:
I'm still parsing through the rest of the code but I would recommend changing thing from a singleton to a normal instance and registering it with the current environment. See PR #49 for an idea of what I did to the numbering singleton.